frontend/src/App.tsx

```diff
@@ -6,19 +6,35 @@ import TodoList from "./components/TodoList";
 import { Todo } from "./models/todo.model";
 
 function App() {
-  const [todos, setTodos] = useState<Todo[]>([
-    { description: "Hello world", id: "244454", isDone: false },
-    { description: "Hello Pakistan", id: "24454", isDone: false },
-    { description: "Hello GB", id: "24433", isDone: false },
-  ]);
-  const [newTodo, setNewTodo] = useState<Todo>({
-    description: "Hello world",
-    id: "244454",
-    isDone: false,
-  });
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [newTodoDescription, setNewTodoDescription] = useState("");
 
   const onChange = (name: string, value: string) => {
-    console.log(name, ":", value);
+    setNewTodoDescription(value);
+  };
+
+  const addTask = () => {
+    if (newTodoDescription.trim().length === 0) {
+      alert("Enter valid description!");
+    } else {
+      fetch("http://localhost:3001/todos", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ description: newTodoDescription }),
+      })
+        .then((response) => response.json())
+        .then((data) => {
+          console.log("Success:", data);
+          const newtodos: Todo[] = [...todos, data];
```

Comment:
There is an issue with this method of updating state. It seems to work fine in this case, but actually this is the wrong way. There is a different syntax for setting new state, if the new state is dependent on the previous state. You have used that method before, in a different project.

Update the state here using that method.